### DISCUSSION TITLE
How to use Dynamic triggers ?

### DISCUSSION THREAD

**Discussion by MatthieuGonnet** [buildbot/buildbot#6555] [Q&A]
Hello everyone !

Thanks for maintaining this amazing project !
I wanted to knpw how to use dynamic triggers ?

I defined several triggerable schedulers, and I want to trigger some of them depending on a property of my build.

```
f.addStep(steps.SetPropertyFromCommand(command=f"make updeps-auto-{repo} FLAVOR={elementoption}", extract_fn=updeps_list))       

#The idea here is to use something like that
f.addStep(steps.Trigger(schedulerNames=util.Interpolate("%(prop:updeps_list)s"),
                                       waitForFinish=True,
))
```

Do I need to override the Trigger class ? I read something about dynamic Triggers but I don't know how to use them https://docs.buildbot.net/current/manual/configuration/steps/trigger.html#dynamic-trigger.

PS: my idea is not to use virtual builders

Thanks !!

**Comment by vibbo** [comment]
Hi, try

```
f.addStep(steps.Trigger(schedulerNames=[util.Interpolate("%(prop:updeps_list)s")],
                                       waitForFinish=True,
))
```
`schedulerNames` is indeed a renderable, but it is expected to be a list of renderables:
https://github.com/buildbot/buildbot/blob/master/master/buildbot/steps/trigger.py#L73-L83
So your example probably fails at the attempt to list over a renderable.

I haven't actually tested it so take it with a grain of salt.

martin
